projects/dsp-ui/src/lib/core/session.service.ts

```diff
@@ -0,0 +1,215 @@
+import { Inject, Injectable } from '@angular/core';
+import {
+    ApiResponseData,
+    ApiResponseError,
+    Constants,
+    CredentialsResponse,
+    KnoraApiConfig,
+    KnoraApiConnection,
+    UserResponse
+} from '@dasch-swiss/dsp-js';
+import { DspApiConfigToken, DspApiConnectionToken } from './core.module';
+
+/**
+ * Currently logged-in user information
+ */
+export interface CurrentUser {
+    // username
+    name: string;
+
+    // json web token
+    jwt: string;
+
+    // default language for ui
+    lang: string;
+
+    // is system admin?
+    sysAdmin: boolean;
+
+    // list of project shortcodes where the user is project admin
+    projectAdmin: string[];
+}
+
+/**
+ * Session with id (= login timestamp) and inforamtion about logged-in user
+ */
+export interface Session {
+    id: number;
+    user: CurrentUser;
+}
+
+@Injectable({
+    providedIn: 'root'
+})
+export class SessionService {
+
+    /**
+     * max session time in milliseconds
+     * default value (24h): 86400000
+     *
+     */
+    readonly MAX_SESSION_TIME: number = 86400000; // 1d = 24 * 60 * 60 * 1000
+
+
+    constructor(
+        @Inject(DspApiConnectionToken) private dspApiConnection: KnoraApiConnection,
+        @Inject(DspApiConfigToken) private dspApiConfig: KnoraApiConfig
+    ) { }
+
+    /**
+     * get session information from localstorage
+     */
+    getSession(): Session {
```

Comment:
I would be better to give this function an explicit return type. If you cannot find user information, return `null`, e.g. `Session | null`.

Reply:
That's true.

Reply:
Resolved in cb139e0